common/src/services/state.service.ts

```diff
@@ -1,27 +1,1377 @@
+        if (this.state.accounts == null) {
+            return null;
+        }
+        return this.state.accounts[this.getUserIdFromMemory(options)];
+    }
+
+    private getUserIdFromMemory(options: StorageOptions): string {
+        return options?.userId != null ?
+            this.state.accounts[options.userId].userId :
+            this.state.activeUserId;
+    }
+
+    private async getAccountFromDisk(options: StorageOptions): Promise<Account> {
+        if (options?.userId == null && this.state.activeUserId == null) {
+            return null;
+        }
+
+        const account = options?.useSecureStorage ?
+            await this.secureStorageService.get<Account>(options.userId ?? this.state.activeUserId, options) :
+            await this.storageService.get<Account>(options.userId ?? this.state.activeUserId, options);
+        return account;
+    }
+
+    private useMemory(storageLocation: StorageLocation) {
+        return storageLocation === StorageLocation.Memory ||
+            storageLocation === StorageLocation.Both;
+    }
+
+    private useDisk(storageLocation: StorageLocation) {
+        return storageLocation === StorageLocation.Disk ||
+            storageLocation === StorageLocation.Both;
+    }
+
+    private async saveAccount(account: Account, options: StorageOptions = {
+        storageLocation: StorageLocation.Both,
+        useSecureStorage: false,
+    }) {
+        return this.useMemory(options.storageLocation) ?
+            await this.saveAccountToMemory(account) :
+            await this.saveAccountToDisk(account, options);
     }
 
-    save(key: string, obj: any): Promise<any> {
-        this.state[key] = obj;
-        return Promise.resolve();
+    private async saveAccountToDisk(account: Account, options: StorageOptions): Promise<void> {
+        if (options.useSecureStorage) {
+            await this.secureStorageService.save(account.userId, account, options);
+        } else {
+            await this.storageService.save(account.userId, account, options);
+        }
+        await this.pushAccounts();
+    }
+
+    private async saveAccountToMemory(account: Account): Promise<void> {
+        if (this.getAccountFromMemory({ userId: account.userId }) !== null) {
+            this.state.accounts[account.userId] = account;
+        }
+        await this.pushAccounts();
+    }
+
+    private async scaffoldNewAccountStorage(account: Account): Promise<void> {
+        const storageAccount = await this.storageService.get<Account>(account.userId);
+        if (storageAccount != null) {
+            storageAccount.accessToken = account.accessToken;
+            storageAccount.refreshToken = account.refreshToken;
+            account = storageAccount;
+        }
+        await this.storageService.save(account.userId, account);
+
+        if (await this.secureStorageService.get<Account>(account.userId) == null) {
+            await this.secureStorageService.save(account.userId, account);
```

Comment:
We talked about this in slack in context of web, but we should make sure that each storage service implemented in the different projects has the account accessible for each of the `StorageOptions`. For example, how web defaults to storing things in Session Storage, but account is attempted to be accessed in local storage. An example of this is in `getTheme`

Reply:
@addisonbeck Looks like your latest commit addresses this!